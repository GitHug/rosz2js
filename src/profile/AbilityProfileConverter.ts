import AbstractProfileConverter from './AbstractProfileConverter';
import { AbilityProfile, BSAbilitiesCharacteristic, TypeName } from '../types';

class AbilityProfileConverter extends AbstractProfileConverter<AbilityProfile, BSAbilitiesCharacteristic> {
  constructor() {
    super({
      typeName: TypeName.ABILITY,
      name: '-',
      description: '-'
    });
  }

  protected getProperty(bsCharacteristic: BSAbilitiesCharacteristic): Partial<AbilityProfile> {
    return bsCharacteristic.$.name === 'Description' ? { description: bsCharacteristic._ } : {};
  }
}

export default AbilityProfileConverter;
