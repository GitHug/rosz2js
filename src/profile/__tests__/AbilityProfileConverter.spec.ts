import AbilityProfileConverter from '../AbilityProfileConverter';
import { BSTypeName, TypeName } from '../../Types';

describe('AbilityProfileConverter', () => {
  it('should convert ability characteristics', () => {
    const converter = new AbilityProfileConverter();

    const profile = converter.convert({
      $: {
        typeName: BSTypeName.ABILITIES,
        name: 'Super Ability'
      },
      characteristics: [
        {
          characteristic: [
            {
              $: {
                name: 'Description'
              },
              _: 'Extra Ability'
            }
          ]
        }
      ]
    });

    expect(profile).toEqual({
      typeName: TypeName.ABILITY,
      name: 'Super Ability',
      description: 'Extra Ability'
    });
    expect(profile).toMatchSnapshot();
  });

  it('should have default values', () => {
    const converter = new AbilityProfileConverter();

    const profile = converter.convert({
      $: {
        typeName: BSTypeName.ABILITIES,
        name: 'Super Ability'
      },
      characteristics: []
    });

    expect(profile).toEqual({
      typeName: TypeName.ABILITY,
      name: 'Super Ability',
      description: '-'
    });
    expect(profile).toMatchSnapshot();
  });
});
