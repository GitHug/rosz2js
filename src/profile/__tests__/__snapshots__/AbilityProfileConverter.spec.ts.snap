// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbilityProfileConverter should convert ability characteristics 1`] = `
Object {
  "description": "Extra Ability",
  "name": "Super Ability",
  "typeName": "Ability",
}
`;

exports[`AbilityProfileConverter should have default values 1`] = `
Object {
  "description": "-",
  "name": "Super Ability",
  "typeName": "Ability",
}
`;
